File: plugins/xtest/examples/junit4/MoneyTest.java
import java.lang.Exception;
import org.junit.Test;
import static org.junit.Assert.assertEquals;

public class MoneyTest {
    @Test
    public void create() throws Exception {
        Money m = new Money(5, "USD");
        assertEquals(5, m.amount());
        assertEquals("USD", m.currency());
    }

    @Test(expected = Money.NegativeAmountException.class)
    public void negative() throws Exception {
         new Money(-5, "FOO");
    }

    @Test
    public void zero() throws Exception {
        Money m = new Money(0, "USD");
        assertEquals(0, m.amount());
    }
}

File: plugins/xtest/examples/nunit/moneytest.cs
using NUnit.Framework;

[TestFixture]
public class MoneyTest {

    [Test]
    public void create() {
	Money m = new Money(5, "USD");
	Assert.AreEqual(5, m.amount());
	Assert.AreEqual("USD", m.currency());
    }

    [Test]
    public void negative() {
	try {
	    new Money(-5, "USD");
	    Assert.Fail("Should throw NegativeAmountException");
	} catch (NegativeAmountException e) {
	}
    }

    [Test]
    public void zero() {
	Money m = new Money(0, "USD");
	Assert.AreEqual(0, m.amount());
    }
}
